# Data Cleaning

In this jupyter notebook, we aim to clean the main dataset in 3 ways so as to obtain the an optimal set of data that can be used to train our machine learning models. 

NOTE: 
The output of this notebook serves as a baseline csv cleaned data. Further data cleaning might be done after running the dataset with our machine learning models

Ways to clean the data:
1. Removing year(s) 
2. Removing variables that have too many missing values 
3. Removing countries that have too many missing values

In [1]:
import pandas as pd
import csv
import numpy as np

asia_dataset = pd.read_csv('asia_continent_dataset.csv')
pd.set_option('display.max_rows', asia_dataset.shape[0]+1)
pd.set_option('display.max_columns', None)

In [2]:
print("Shape: " + str(asia_dataset.shape))
asia_dataset

Shape: (200, 28)


Countries  Year  GDP_Growth  Market_Size  Life_Exp  \
0           Australia  2019    2.160956    52203.130  82.90000   
1           Australia  2018    2.949286    50252.840  82.74878   
2           Australia  2017    2.300611    48482.650  82.50000   
3           Australia  2016    2.770652    47339.970  82.44878   
4           Australia  2015    2.192647    46288.330  82.40000   
5           Australia  2014    2.533115    46880.220  82.30000   
6           Australia  2013    2.584898    45902.050  82.14878   
7           Australia  2012    3.918163    42826.790  82.04634   
8           Australia  2011    2.462756    41965.360  81.89512   
9           Australia  2010    2.067417    39301.340  81.69512   
10         Bangladesh  2019    8.152685     4954.761  72.59100   
11         Bangladesh  2018    7.863743     4547.480  72.32000   
12         Bangladesh  2017    7.284174     4160.703  72.05200   
13         Bangladesh  2016    7.113478     3848.958  71.78500   
14         Bangladesh  2015    6.552640     3555.505  71.51400   
15         Bangladesh  2014    6.061059     3368.056  71.23100   
16         Bangladesh  2013    6.013606     3143.586  70.93000   
17         Bangladesh  2012    6.521459     2937.629  70.60600   
18         Bangladesh  2011    6.464379     2620.834  70.25600   
19         Bangladesh  2010    5.571788     2439.070  69.88100   
20             Brunei  2019    3.869110    64724.130  75.86000   
21             Brunei  2018    0.052238    61839.120  75.72200   
22             Brunei  2017    1.328603    60994.530  75.58500   
23             Brunei  2016   -2.477920    56297.230  75.45000   
24             Brunei  2015   -0.392380    62540.630  75.31800   
25             Brunei  2014   -2.508350    81371.230  75.19300   
26             Brunei  2013   -2.126030    83788.510  75.07500   
27             Brunei  2012    0.912842    88245.960  74.96200   
28             Brunei  2011    3.745318    83935.230  74.85200   
29             Brunei  2010    2.598966    80280.020  74.74000   
30              China  2019    5.949714    16772.840  76.91200   
31              China  2018    6.749774    15608.970  76.70400   
32              China  2017    6.947201    14344.420  76.47000   
33              China  2016    6.848762    13572.620  76.21000   
34              China  2015    7.041329    12978.770  75.92800   
35              China  2014    7.425764    12549.770  75.62900   
36              China  2013    7.766150    11923.750  75.32100   
37              China  2012    7.863736    11197.600  75.01300   
38              China  2011    9.550832    10299.870  74.70800   
39              China  2010   10.635870     9253.765  74.40900   
40          Hong Kong  2019   -1.679760    62106.100  85.07805   
41          Hong Kong  2018    2.846448    62529.060  84.93415   
42          Hong Kong  2017    3.791006    59849.250  84.68049   
43          Hong Kong  2016    2.169494    57221.410  84.22683   
44          Hong Kong  2015    2.387802    56408.940  84.27805   
45          Hong Kong  2014    2.762377    54781.650  83.98049   
46          Hong Kong  2013    3.101547    53691.840  83.83171   
47          Hong Kong  2012    1.700325    52234.590  83.48049   
48          Hong Kong  2011    4.814682    52277.470  83.42195   
49          Hong Kong  2010    6.767669    49185.230  82.97805   
50              India  2019    4.041554     6997.864  69.65600   
51              India  2018    6.532989     6675.361  69.41600   
52              India  2017    6.795383     6182.922  69.16500   
53              India  2016    8.256306     5839.865  68.89700   
54              India  2015    7.996254     5464.859  68.60700   
55              India  2014    7.410228     5233.882  68.28600   
56              India  2013    6.386106     5057.233  67.93100   
57              India  2012    5.456389     4861.156  67.54500   
58              India  2011    5.241315     4493.669  67.13000   
59              India  2010    8.497585     4236.744

In [3]:
# Getting the list of asian countries 

countries = []

for x in range(len(asia_dataset)):
        if countries.count(asia_dataset.iloc[x,0]) > 0:
            continue
        else: 
            countries.append(asia_dataset.iloc[x,0])

print("Number of countries: " + str(len(countries)))
countries

Number of countries: 20


['Australia',
 'Bangladesh',
 'Brunei',
 'China',
 'Hong Kong',
 'India',
 'Indonesia',
 'Japan',
 'Malaysia',
 'Myanmar',
 'New Zealand',
 'Pakistan',
 'Papua New Guinea',
 'Philippines',
 'Singapore',
 'South Korea',
 'Sri Lanka',
 'Thailand',
 'Timor-Leste',
 'Vietnam']

# Removing Years

Our group has scraped data for years 2010-2019 (10years), and did not scraped for year 2020. This is because of covid-19 that has taken place in year 2020 which have affected many countries severely. This would have caused many of the indicators for year 2020 to be an outlier or "out-of-the-norm" from the previous years. Thus, we have chose to not add year 2020 as it would affect our model's accuracy and performance.

In [4]:
# Creating a dataframe to track the number of NaN per year 
emptyYears = pd.DataFrame( {"Year":[2010,2011,2012,2013,2014,2015,2016,2017,2018,2019]})
emptyYears["Empty_Entries"] = 0
emptyYears

Year  Empty_Entries
0  2010              0
1  2011              0
2  2012              0
3  2013              0
4  2014              0
5  2015              0
6  2016              0
7  2017              0
8  2018              0
9  2019              0

In [5]:
for x in range(len(asia_dataset)):
    for y in range(len(emptyYears)):
        
        if asia_dataset.iloc[x,1] == emptyYears.iloc[y,0]:
            emptyYears.iloc[y,1] = emptyYears.iloc[y,1] + asia_dataset.loc[[x]].isna().sum().sum()

In [6]:
emptyYears

Year  Empty_Entries
0  2010            127
1  2011            119
2  2012            102
3  2013             80
4  2014             78
5  2015             75
6  2016             76
7  2017             74
8  2018             69
9  2019             75

We have a total of 20 countries and 26 variables, thus a total of 520 entries per year. Referring to the cell above, we noticed that year 2010 has the most NaN entries, at a value of 127. However, we will still keep year 2010 as it still has 77% of its data filled, which is still a substantial amount of non-missing data

# Removing Variables


In [7]:
# Creating a dataframe to track the number of NaN per variable
columns_name = list(asia_dataset.columns)

columns_name.remove("Countries")
columns_name.remove("Year")
columns_name

emptyCol = pd.DataFrame( {"Variables":columns_name})
emptyCol["Empty_Entries"] = 0
emptyCol

Variables  Empty_Entries
0                          GDP_Growth              0
1                         Market_Size              0
2                            Life_Exp              0
3             Mean_Years_of_Schooling              0
4                Internet_Penetration              0
5                       Gov_Indicator              0
6                      Trade_Openness              0
7                        Labour_Force              0
8   Domestic_credit_to_private_sector              0
9                           inflation              0
10                            rnd_exp              0
11                                GNI              0
12                                BoT              0
13                               Gini              0
14                                LPI              0
15                    Child_Mortality              0
16                                HDI              0
17                                EPI              0
18                                GCI              0
19                      Pol_Stability              0
20                                CPI              0
21                          Urban_pop              0
22                       Renew_Energy              0
23                      CO2_emissions              0
24                        ind_val_add              0
25                  unemployment_rate              0

In [8]:
for index, name in enumerate(columns_name):
    emptyCol.iloc[index,1] = asia_dataset[name].isna().sum()
    
emptyCol

Variables  Empty_Entries
0                          GDP_Growth              0
1                         Market_Size              0
2                            Life_Exp              0
3             Mean_Years_of_Schooling              0
4                Internet_Penetration             21
5                       Gov_Indicator              0
6                      Trade_Openness             10
7                        Labour_Force              0
8   Domestic_credit_to_private_sector             34
9                           inflation             30
10                            rnd_exp            130
11                                GNI             30
12                                BoT             20
13                               Gini             82
14                                LPI            200
15                    Child_Mortality             10
16                                HDI              0
17                                EPI             80
18                                GCI            165
19                      Pol_Stability              0
20                                CPI             40
21                          Urban_pop              0
22                       Renew_Energy             20
23                      CO2_emissions              0
24                        ind_val_add              3
25                  unemployment_rate              0

We have a total of 200 rows as such, to accurately obtain variables that will add value to our machine learning model, we will only be using variables that have more than 80% of its cells filled. 

Referring to the cell above, we can see that 
1. rnd_exp
2. Gini
3. LPI
4. EPI
5. GCI 
<br> 

all have more than 40 entries that are not filled, as such we will be removing these variables from our main_dataset dataframe. Bringing our total number of variables used down from 26 to 21

In [9]:
asia_dataset = asia_dataset.drop(columns= ['rnd_exp','LPI','EPI','GCI','Gini'])
asia_dataset

Countries  Year  GDP_Growth  Market_Size  Life_Exp  \
0           Australia  2019    2.160956    52203.130  82.90000   
1           Australia  2018    2.949286    50252.840  82.74878   
2           Australia  2017    2.300611    48482.650  82.50000   
3           Australia  2016    2.770652    47339.970  82.44878   
4           Australia  2015    2.192647    46288.330  82.40000   
5           Australia  2014    2.533115    46880.220  82.30000   
6           Australia  2013    2.584898    45902.050  82.14878   
7           Australia  2012    3.918163    42826.790  82.04634   
8           Australia  2011    2.462756    41965.360  81.89512   
9           Australia  2010    2.067417    39301.340  81.69512   
10         Bangladesh  2019    8.152685     4954.761  72.59100   
11         Bangladesh  2018    7.863743     4547.480  72.32000   
12         Bangladesh  2017    7.284174     4160.703  72.05200   
13         Bangladesh  2016    7.113478     3848.958  71.78500   
14         Bangladesh  2015    6.552640     3555.505  71.51400   
15         Bangladesh  2014    6.061059     3368.056  71.23100   
16         Bangladesh  2013    6.013606     3143.586  70.93000   
17         Bangladesh  2012    6.521459     2937.629  70.60600   
18         Bangladesh  2011    6.464379     2620.834  70.25600   
19         Bangladesh  2010    5.571788     2439.070  69.88100   
20             Brunei  2019    3.869110    64724.130  75.86000   
21             Brunei  2018    0.052238    61839.120  75.72200   
22             Brunei  2017    1.328603    60994.530  75.58500   
23             Brunei  2016   -2.477920    56297.230  75.45000   
24             Brunei  2015   -0.392380    62540.630  75.31800   
25             Brunei  2014   -2.508350    81371.230  75.19300   
26             Brunei  2013   -2.126030    83788.510  75.07500   
27             Brunei  2012    0.912842    88245.960  74.96200   
28             Brunei  2011    3.745318    83935.230  74.85200   
29             Brunei  2010    2.598966    80280.020  74.74000   
30              China  2019    5.949714    16772.840  76.91200   
31              China  2018    6.749774    15608.970  76.70400   
32              China  2017    6.947201    14344.420  76.47000   
33              China  2016    6.848762    13572.620  76.21000   
34              China  2015    7.041329    12978.770  75.92800   
35              China  2014    7.425764    12549.770  75.62900   
36              China  2013    7.766150    11923.750  75.32100   
37              China  2012    7.863736    11197.600  75.01300   
38              China  2011    9.550832    10299.870  74.70800   
39              China  2010   10.635870     9253.765  74.40900   
40          Hong Kong  2019   -1.679760    62106.100  85.07805   
41          Hong Kong  2018    2.846448    62529.060  84.93415   
42          Hong Kong  2017    3.791006    59849.250  84.68049   
43          Hong Kong  2016    2.169494    57221.410  84.22683   
44          Hong Kong  2015    2.387802    56408.940  84.27805   
45          Hong Kong  2014    2.762377    54781.650  83.98049   
46          Hong Kong  2013    3.101547    53691.840  83.83171   
47          Hong Kong  2012    1.700325    52234.590  83.48049   
48          Hong Kong  2011    4.814682    52277.470  83.42195   
49          Hong Kong  2010    6.767669    49185.230  82.97805   
50              India  2019    4.041554     6997.864  69.65600   
51              India  2018    6.532989     6675.361  69.41600   
52              India  2017    6.795383     6182.922  69.16500   
53              India  2016    8.256306     5839.865  68.89700   
54              India  2015    7.996254     5464.859  68.60700   
55              India  2014    7.410228     5233.882  68.28600   
56              India  2013    6.386106     5057.233  67.93100   
57              India  2012    5.456389     4861.156  67.54500   
58              India  2011    5.241315     4493.669  67.13000   
59              India  2010    8.497585     4236.744

# Removing Countries

Here we aim to remove countries that have too many missing values 


In [10]:
# Creating a dataframe to track the number of NaN per country
columns = asia_dataset.columns.values.tolist()
columns.remove("Countries")
columns.remove("Year")

emptyCountries = pd.DataFrame( {"Countries": countries})
for column in columns:
     emptyCountries[column] = 0
emptyCountries

Countries  GDP_Growth  Market_Size  Life_Exp  \
0          Australia           0            0         0   
1         Bangladesh           0            0         0   
2             Brunei           0            0         0   
3              China           0            0         0   
4          Hong Kong           0            0         0   
5              India           0            0         0   
6          Indonesia           0            0         0   
7              Japan           0            0         0   
8           Malaysia           0            0         0   
9            Myanmar           0            0         0   
10       New Zealand           0            0         0   
11          Pakistan           0            0         0   
12  Papua New Guinea           0            0         0   
13       Philippines           0            0         0   
14         Singapore           0            0         0   
15       South Korea           0            0         0   
16         Sri Lanka           0            0         0   
17          Thailand           0            0         0   
18       Timor-Leste           0            0         0   
19           Vietnam           0            0         0   

    Mean_Years_of_Schooling  Internet_Penetration  Gov_Indicator  \
0                         0                     0              0   
1                         0                     0              0   
2                         0                     0              0   
3                         0                     0              0   
4                         0                     0              0   
5                         0                     0              0   
6                         0                     0              0   
7                         0                     0              0   
8                         0                     0              0   
9                         0                     0              0   
10                        0                     0              0   
11                        0                     0              0   
12                        0                     0              0   
13                        0                     0              0   
14                        0                     0              0   
15                        0                     0              0   
16                        0                     0              0   
17                        0                     0              0   
18                        0                     0              0   
19                        0                     0              0   

    Trade_Openness  Labour_Force  Domestic_credit_to_private_sector  \
0                0             0                                  0   
1                0             0                                  0   
2                0             0                                  0   
3                0             0                                  0   
4                0             0                                  0   
5                0             0                                  0   
6                0             0                                  0   
7                0             0                                  0   
8                0             0                                  0   
9                0             0                                  0   
10               0             0                                  0   
11               0             0                                  0   
12               0             0                                  0   
13               0             0                                  0   
14               0             0                                  0   
15               0             0                                  0   
16               0             0                                  0   
17               0             0                                

In [11]:
for x in range(len(emptyCountries)):
    for y in range(len(asia_dataset)):
        
        if asia_dataset.iloc[y,0] == emptyCountries.iloc[x,0]:
            for index, name in enumerate(columns):
                if pd.isna(asia_dataset.iloc[y,index+2]) == True:
                    emptyCountries.iloc[x,index+1] = emptyCountries.iloc[x,index+1] + 1
                    #emptyCountries.iloc[y,index] = emptyCountries.iloc[y,index] + asia_dataset.loc[[x]].isna().sum().sum()

In [12]:
emptyCountries

Countries  GDP_Growth  Market_Size  Life_Exp  \
0          Australia           0            0         0   
1         Bangladesh           0            0         0   
2             Brunei           0            0         0   
3              China           0            0         0   
4          Hong Kong           0            0         0   
5              India           0            0         0   
6          Indonesia           0            0         0   
7              Japan           0            0         0   
8           Malaysia           0            0         0   
9            Myanmar           0            0         0   
10       New Zealand           0            0         0   
11          Pakistan           0            0         0   
12  Papua New Guinea           0            0         0   
13       Philippines           0            0         0   
14         Singapore           0            0         0   
15       South Korea           0            0         0   
16         Sri Lanka           0            0         0   
17          Thailand           0            0         0   
18       Timor-Leste           0            0         0   
19           Vietnam           0            0         0   

    Mean_Years_of_Schooling  Internet_Penetration  Gov_Indicator  \
0                         0                     2              0   
1                         0                     0              0   
2                         0                     0              0   
3                         0                     0              0   
4                         0                     0              0   
5                         0                     0              0   
6                         0                     0              0   
7                         0                     0              0   
8                         0                     0              0   
9                         0                     4              0   
10                        0                     2              0   
11                        0                     0              0   
12                        0                     2              0   
13                        0                     5              0   
14                        0                     0              0   
15                        0                     0              0   
16                        0                     4              0   
17                        0                     0              0   
18                        0                     2              0   
19                        0                     0              0   

    Trade_Openness  Labour_Force  Domestic_credit_to_private_sector  \
0                0             0                                  0   
1                0             0                                  0   
2                0             0                                 10   
3                0             0                                  0   
4                0             0                                 10   
5                0             0                                  0   
6                0             0                                  0   
7                0             0                                  0   
8                0             0                                  0   
9                0             0                                  0   
10               0             0                                  4   
11               0             0                                  0   
12              10             0                                  0   
13               0             0                                  0   
14               0             0                                  0   
15               0             0                                 10   
16               0             0                                  0   
17               0             0                                

In [13]:
columns_to_drop = []

for index, name in enumerate(columns):
    total = 0 
    for x in range(len(emptyCountries)):
        total = total + int(emptyCountries.iloc[x,index+1]) 
    if total == 0:
           columns_to_drop.append(name)
            
columns_to_drop

['GDP_Growth',
 'Market_Size',
 'Life_Exp',
 'Mean_Years_of_Schooling',
 'Gov_Indicator',
 'Labour_Force',
 'HDI',
 'Pol_Stability',
 'Urban_pop',
 'CO2_emissions',
 'unemployment_rate']

In [14]:
emptyCountries = emptyCountries.drop(columns= columns_to_drop)
emptyCountries

Countries  Internet_Penetration  Trade_Openness  \
0          Australia                     2               0   
1         Bangladesh                     0               0   
2             Brunei                     0               0   
3              China                     0               0   
4          Hong Kong                     0               0   
5              India                     0               0   
6          Indonesia                     0               0   
7              Japan                     0               0   
8           Malaysia                     0               0   
9            Myanmar                     4               0   
10       New Zealand                     2               0   
11          Pakistan                     0               0   
12  Papua New Guinea                     2              10   
13       Philippines                     5               0   
14         Singapore                     0               0   
15       South Korea                     0               0   
16         Sri Lanka                     4               0   
17          Thailand                     0               0   
18       Timor-Leste                     2               0   
19           Vietnam                     0               0   

    Domestic_credit_to_private_sector  inflation  GNI  BoT  Child_Mortality  \
0                                   0          0    0    0                0   
1                                   0          0    0    0                0   
2                                  10         10   10    0                0   
3                                   0          0    0    0                0   
4                                  10         10   10    0               10   
5                                   0          0    0    0                0   
6                                   0          0    0    0                0   
7                                   0          0    0    0                0   
8                                   0          0    0    0                0   
9                                   0          0    0    0                0   
10                                  4          0    0    0                0   
11                                  0          0    0    0                0   
12                                  0          0    0   10                0   
13                                  0          0    0    0                0   
14                                  0          0    0    0                0   
15                                 10         10   10    0                0   
16                                  0          0    0    0                0   
17                                  0          0    0    0                0   
18                                  0          0    0   10                0   
19                                  0          0    0    0                0   

    CPI  Renew_Energy  ind_val_add  
0     2             0            0  
1     2             0            0  
2     2             0            0  
3     2             0            0  
4     2             0            0  
5     2             0            0  
6     2             0            0  
7     2             0            1  
8     2             0            0  
9     2            10            0  
10    2             0            1  
11    2             0            0  
12    2             0            1  
13    2             0            0  
14    2             0            0  
15    2             0            0  
16    2             0            0  
17    2             0            0  
18    2            10            0  
19    2             0            0

For each country, there is 10 years worth of data. To ensure that we do not have too many consecutive missing values, we aim to remove those countries that have more than 2 variables with 10 years worth of missing data.

Referring to the cell above, we will remove 
1. Brunei
2. Hong Kong
3. Papua New Guinea
3. South Korea
3. Timor-Leste
<br>

bringing our total countries down from 20 to 15.

In [15]:
asia_dataset = asia_dataset.drop(asia_dataset[asia_dataset.Countries == "Brunei"].index)
asia_dataset = asia_dataset.drop(asia_dataset[asia_dataset.Countries == "Hong Kong"].index)
asia_dataset = asia_dataset.drop(asia_dataset[asia_dataset.Countries == "South Korea"].index)
asia_dataset = asia_dataset.drop(asia_dataset[asia_dataset.Countries == "Timor-Leste"].index)
asia_dataset = asia_dataset.drop(asia_dataset[asia_dataset.Countries == "Papua New Guinea"].index)

asia_dataset = asia_dataset.reset_index(drop=True)

In [16]:
print("Shape: " + str(asia_dataset.shape))
asia_dataset

Shape: (150, 23)


Countries  Year  GDP_Growth  Market_Size  Life_Exp  \
0      Australia  2019    2.160956    52203.130  82.90000   
1      Australia  2018    2.949286    50252.840  82.74878   
2      Australia  2017    2.300611    48482.650  82.50000   
3      Australia  2016    2.770652    47339.970  82.44878   
4      Australia  2015    2.192647    46288.330  82.40000   
5      Australia  2014    2.533115    46880.220  82.30000   
6      Australia  2013    2.584898    45902.050  82.14878   
7      Australia  2012    3.918163    42826.790  82.04634   
8      Australia  2011    2.462756    41965.360  81.89512   
9      Australia  2010    2.067417    39301.340  81.69512   
10    Bangladesh  2019    8.152685     4954.761  72.59100   
11    Bangladesh  2018    7.863743     4547.480  72.32000   
12    Bangladesh  2017    7.284174     4160.703  72.05200   
13    Bangladesh  2016    7.113478     3848.958  71.78500   
14    Bangladesh  2015    6.552640     3555.505  71.51400   
15    Bangladesh  2014    6.061059     3368.056  71.23100   
16    Bangladesh  2013    6.013606     3143.586  70.93000   
17    Bangladesh  2012    6.521459     2937.629  70.60600   
18    Bangladesh  2011    6.464379     2620.834  70.25600   
19    Bangladesh  2010    5.571788     2439.070  69.88100   
20         China  2019    5.949714    16772.840  76.91200   
21         China  2018    6.749774    15608.970  76.70400   
22         China  2017    6.947201    14344.420  76.47000   
23         China  2016    6.848762    13572.620  76.21000   
24         China  2015    7.041329    12978.770  75.92800   
25         China  2014    7.425764    12549.770  75.62900   
26         China  2013    7.766150    11923.750  75.32100   
27         China  2012    7.863736    11197.600  75.01300   
28         China  2011    9.550832    10299.870  74.70800   
29         China  2010   10.635870     9253.765  74.40900   
30         India  2019    4.041554     6997.864  69.65600   
31         India  2018    6.532989     6675.361  69.41600   
32         India  2017    6.795383     6182.922  69.16500   
33         India  2016    8.256306     5839.865  68.89700   
34         India  2015    7.996254     5464.859  68.60700   
35         India  2014    7.410228     5233.882  68.28600   
36         India  2013    6.386106     5057.233  67.93100   
37         India  2012    5.456389     4861.156  67.54500   
38         India  2011    5.241315     4493.669  67.13000   
39         India  2010    8.497585     4236.744  66.69300   
40     Indonesia  2019    5.018160    12311.500  71.71600   
41     Indonesia  2018    5.174292    11644.760  71.50900   
42     Indonesia  2017    5.069786    10935.630  71.28200   
43     Indonesia  2016    5.033069    10494.470  71.03500   
44     Indonesia  2015    4.876322    10247.210  70.76800   
45     Indonesia  2014    5.006668    10278.180  70.48100   
46     Indonesia  2013    5.557264    10067.470  70.17900   
47     Indonesia  2012    6.030051     9713.900  69.86600   
48     Indonesia  2011    6.169784     9095.742  69.54200   
49     Indonesia  2010    6.223854     8505.748  69.20500   
50         Japan  2019    0.270309    42197.250  84.35634   
51         Japan  2018    0.323207    41695.640  84.21098   
52         Japan  2017    2.168291    40966.590  84.09976   
53         Japan  2016    0.521944    39964.280  83.98488   
54         Japan  2015    1.222921    40388.550  83.79390   
55         Japan  2014    0.374719    39179.160  83.58780   
56         Japan  2013    2.000268    38974.080  83.33195   
57         Japan  2012    1.495090    37191.390  83.09610   
58         Japan  2011   -0.115420    35774.700  82.59122   
59         Japan  2010    4.191739    34986.990  82.84268   
60      Malaysia  2019    4.302816    29564.010  76.15600   
61      Malaysia  2018    4.769927    28219.800  75.99700   
62      Malaysia  2017    5.812722    26661.510  75.82800   
63      Malaysia  2016    4.449781    25546.130  75.64900   
64      Malaysia  2015    5.091532    2480

# Checking Datatypes

In this section, we aim to check the datatypes of the columns and if need, convert them into the corrrect datatype for proper use



In [17]:
asia_dataset.dtypes

Countries                             object
Year                                   int64
GDP_Growth                           float64
Market_Size                          float64
Life_Exp                             float64
Mean_Years_of_Schooling              float64
Internet_Penetration                 float64
Gov_Indicator                        float64
Trade_Openness                       float64
Labour_Force                         float64
Domestic_credit_to_private_sector    float64
inflation                            float64
GNI                                  float64
BoT                                  float64
Child_Mortality                      float64
HDI                                  float64
Pol_Stability                        float64
CPI                                   object
Urban_pop                            float64
Renew_Energy                         float64
CO2_emissions                        float64
ind_val_add                          float64
unemployme

In [18]:
asia_dataset[["CPI"]] = asia_dataset[["CPI"]].apply(pd.to_numeric)
asia_dataset.dtypes

Countries                             object
Year                                   int64
GDP_Growth                           float64
Market_Size                          float64
Life_Exp                             float64
Mean_Years_of_Schooling              float64
Internet_Penetration                 float64
Gov_Indicator                        float64
Trade_Openness                       float64
Labour_Force                         float64
Domestic_credit_to_private_sector    float64
inflation                            float64
GNI                                  float64
BoT                                  float64
Child_Mortality                      float64
HDI                                  float64
Pol_Stability                        float64
CPI                                  float64
Urban_pop                            float64
Renew_Energy                         float64
CO2_emissions                        float64
ind_val_add                          float64
unemployme

In [19]:
asia_dataset.to_excel('final_asia_dataset_cleaned.xlsx')